### PR TITLE
Improve tests a bit

### PR DESCRIPTION
**Pull request by Hoverbear**
The tests in `Server` looked like they needed some fleshing out. I modified the `get_unbound_address()` to walk up the port numbers until it finds one that can be used.

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -417,26 +417,46 @@ mod test {
 
     /// Attempts to grab a local, unbound socket address for testing.
     fn get_unbound_address() -> SocketAddr {
-        TcpListener::bind("127.0.0.1:0").unwrap().local_addr().unwrap()
+        let mut current = 4001;
```
Comment (danburkert) [inline src/server.rs]: Was there an issue with the way it was implemented before?

File: src/server.rs [hunk 2]
```diff
@@ -417,26 +417,46 @@ mod test {
...
     pub fn test_illegal_peer_address() {
         let _ = env_logger::init();
         let peer_id = ServerId::from(1);
         let mut peers = HashMap::new();
-        peers.insert(peer_id, SocketAddr::from_str("127.0.0.1:0").unwrap());
-        assert!(new_test_server(peers).is_err());
+        peers.insert(peer_id, SocketAddr::from_str("127.0.0.1:4000").unwrap());
+        assert!(new_test_server(
+            ServerId::from(0),
+            SocketAddr::from_str("127.0.0.1:4000").unwrap(),
```
Comment (danburkert) [inline src/server.rs]: This is a bad practice because tests will spuriously fail if the port is bound on the machine.

File: src/server.rs (hunk 1 above)
Comment (Hoverbear) [inline src/server.rs]: Hm, perhaps I misunderstood the goal of the function? I thought it was intended to get a free address.

Comment (danburkert) [inline src/server.rs]: That is correct, binding to port 0 will automatically give a free address picked by the OS.

Comment (Hoverbear) [inline src/server.rs]: Today I learned.

**Pull request closed without merging**